### PR TITLE
2583 serverless blog 1

### PR DESCRIPTION
**Pull request by dmuelle**
#2583

### CODE REVIEW

**Comment by dmuelle** [comment]
https://blogs-draft-openlibertyio.mqj6zf7jocq.us-south.codeengine.appdomain.cloud/blog/2022/10/04/microprofile-serverless-ibm-code-engine.html

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (dmuelle) [inline posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]:
```suggestion
If using the Getting Started code and you already have experience with OpenLiberty and MicroProfile, you can use the `finish` folder as your starting point and go to the next step. If you do not, complete the guide up to and including the https://openliberty.io/guides/getting-started.html#running-the-application-in-a-docker-container[Running the application in a Docker container] step to generate the application you'll use in this post.
```

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (dmuelle) [inline posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]: consider just using the getting started guide for this post- trying to allow for someone to byo app makes the task flow more complicated, and we don't really give enough info. If someone wants to use another app, they'd need to understand well enough how to do it anyway. You can mention that it would work with any RESTful application but still focus the task and description just on using the Getting Started guide app.

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (dmuelle) [inline posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]: here also, you might simplify by focusing on just ICCR, mention the other options but dont discuss

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (dmuelle) [inline posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]:
```suggestion
Listening port::
```

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (dmuelle) [inline posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]:
```suggestion
To call the application on ICCE, we use the same path that we used for the application locally. Run the following `curl` command
```

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (dmuelle) [inline posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc:125]:
```suggestion
```

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (dmuelle) [inline posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]: this part is confusing- I'm not sure Im reading it right but it seems like first we tell them to get version 2.5.0 as a dependency from Maven, but then its says that 2.5.0 isn't available yet and the need to get it from GH and upload to a Maven repo themselves? Is there another version they can use that is available? Or do they have to have to the jakarta package to complete the task?

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (dmuelle) [inline posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]: consider moving the example up here- since the blogs enable scrolling within a codeblock, it wont take up too much page space and having to scroll to the bottom over and over is not ideal. I think readers will want to refer to this example throughout the steps and the snippets can be tough to orient

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (dmuelle) [inline posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]:
```suggestion
First, update the `SystemResource` class response Type from `Response` to `CloudEvent`. In the method declaration, replace `Response` with `CloudEvent`:
```

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (dmuelle) [inline posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc:292]: why would they be missing required properties if they are following these steps?

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (dmuelle) [inline posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc:303]: ^^ what is this? Does the sentence above the code describe what the code is doing?

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (dmuelle) [inline posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]: ^^ something seems off with this sentence but I'm not sure of the intended meaning

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (dmuelle) [inline posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc:338]: ^^ same here, not sure what to suggest but it looks like two sentences running together

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (dmuelle) [inline posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]: returned what?

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (dmuelle) [inline posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]: new version of what?

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (dmuelle) [inline posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]: which previous curl commands?

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (dmuelle) [inline posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc:407]: I think you need at least a sentence or two about the results.. something like

> You now have a MicroProfile application on Open Liberty that runs as a serverless function in IBM Code Engine"

or whatever appropriately summarizes the result

**Review by dmuelle — COMMENTED** [review]
ID review- left a few questions- I think there are probably still some typos to fix but I want to wait until I can review the rebuilt post on draft

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (abutch3r) [reply posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc:292]: This is purely a note to the developer for when they are doing it in their own application what would happen. Would a better way of providing this information be: the Sample above provides only the required properties for a valid CloudEvent. If any of these properties were to be missing then an exception would be thrown

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (abutch3r) [reply posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]: So we recently merged the code changes in to CloudEvents SDK main and we do not yet have a date for 2.5.0 release. the idea here was that when 2.5.0 is released, we would update the blog by simply removing this section. 

The idea is that they can still achieve the task with EE 9+ despite the package not yet being publically available, by building locally and having built locally what other changes they would need to make.

The section about the packages to use is to cover if they wanted to use EE 8 because that is what there App was using then the information they need is there, though would probably require a different Note to cover the change of version to 2.4.0

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (Emily-Jiang) [reply posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]: agreed. Just update it: In this blog post, we will use the Getting Started Guide.

Comment (dmuelle) [reply posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]: updated

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (dmuelle) [reply posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]: updated

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (dmuelle) [reply posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc:292]: updated

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (dmuelle) [reply posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc:407]: if it's relevant, you might also refer them back to the ICCE console to see the application scale when curl requests come in

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (dmuelle) [reply posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]: ok- that makes sense but I wonder if we can simplify the path for the reader so they only worry about one package.

Would the getting started app even work with 2.4.0? I believe that app uses EE 9.1 back dont know if it's back compatible. 

If it does work with 2.4.0, we might just tell them to use that. We can write in the instructions for 2.5 and just comment them out until the release is available.

If the getting started app only works with 2.5.0, then we should tell them how to get that and build it locally, then simplify the instructions whenever the update is released.

Comment (abutch3r) [reply posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]: The alternative package is applicable to MicroProfile 4.0. Microprofile 5+ requires the new package. To get that to work with 4.0 I think requires updating 5 files. So i will work on simplifying that part on just focussing on build locally, and can then update later when the package is formally released.

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (abutch3r) [reply posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc:303]: Have updated the description of what was here to make it clearer what the payload that is sent in request does

File: posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc
Comment (dmuelle) [inline posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]: this is better- but it should mention/reiterate  "serverless" in the result. Anything we can say/show to sum up the value  added by going through this process. Right now it is focused on the tools, how can we emphasize the outcome or benefits?

Comment (abutch3r) [reply posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]: I agree, was working on an alternative

its a bit of a mouthful, but this covers everything that they has been done throughout the post:
`You now have a MicroProfile application on Open Liberty running as a serverless function in IBM Cloud Code Engine with CloudEvents!`

Comment (dmuelle) [reply posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]: Thanks that looks good. With this outcome- can/does the ICCE allocate resources to the app on demand scale it to zero when not in use?

Comment (abutch3r) [reply posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]: The App Startup and  Scale to Zero are covered at the end of the initial App deployment section, before they move onto CloudEvents. given we are also trying to promote ICCE, I think this would be good to move to the end so that it is not lost in the rest of the post.

Comment (dmuelle) [reply posts/2022-10-04-microprofile-serverless-ibm-code-engine.adoc]: great idea

**Comment by dmuelle** [comment]
@abutch3r @Emily-Jiang I think this is in good shape now- thanks for your hard work on it! We need to change the filename to reflect the new publishing date. When would you like to publish? I want to take one last look through the rebuilt staging draft once this PR is merged but I think it could be ready to publish as soon as today or tomorrow.

**Review by abutch3r — APPROVED** [review]

**Comment by abutch3r** [comment]
@dmuelle either today or tomorrow works for us, have approved from my side

**Pull request merged**
